cli/algorithms.go

```diff
@@ -39,13 +47,35 @@ func (cli *CLI) NewAlgorithmCmd() *cobra.Command {
 
 			pemBlock, _ := pem.Decode(privKeyFile)
 
-			privKey, err := x509.ParsePKCS1PrivateKey(pemBlock.Bytes)
-			if err != nil {
-				log.Fatalf("Error parsing private key: %v", err)
-			}
+			bytes, err := x509.ParsePKCS8PrivateKey(pemBlock.Bytes)
```

Comment:
check the type of key returned here rather than errors, see docs here https://pkg.go.dev/crypto/x509#ParsePKCS8PrivateKey